openmc/model/funcs.py

```diff
@@ -110,6 +111,140 @@ def borated_water(boron_ppm, temperature=293., pressure=0.1013, temp_unit='K',
     out.add_s_alpha_beta('c_H_in_H2O')
     return out
 
+def cylindrical_prism(r, height, axis='z', origin=(0., 0., 0.),
+                      boundary_type='transmission',
+                      upper_edge_radius=0.,
+                      lower_edge_radius=0.):
+    """Get a finite cylindrical prism.
+
+    Parameters
+    ----------
+    r: float
+        Prism radius in units of cm.
+    height: float
+        Prism height in units of cm. The height is aligned with the x, y,
+        or z axes for prisms parallel to the x, y, or z axis, respectively.
+    axis : {'x', 'y', 'z'}
+        Axis with which the infinite length of the prism should be aligned.
+        Defaults to 'z'.
+    origin: Iterable of three floats
+        Origin of the prism. The three floats correspond to (x,y,z)
+        Defaults to (0., 0., 0.).
+    boundary_type : {'transmission, 'vacuum', 'reflective', 'periodic'}
+        Boundary condition that defines the behavior for particles hitting the
+        surfaces comprising the rectangular prism (default is 'transmission').
+    upper_edge_radius: float
+        Prism upper edge radius in units of cm. Defaults to 0.
+    lower_edge_radius: float
+        Prism lower edge radius in units of cm. Defaults to 0.
+
+    Returns
+    -------
+    openmc.Region
+        The inside of a rectangular prism
+
+    """
+
+
+    check_type('r', r, Real)
+    check_type('height', height, Real)
+    check_type('upper_edge_radius', upper_edge_radius, Real)
+    check_less_than('upper_edge_radius', upper_edge_radius, r, equality=True)
+    check_type('lower_edge_radius', lower_edge_radius, Real)
+    check_less_than('lower_edge_radius', lower_edge_radius, r, equality=True)
+    check_value('axis', axis, ['x', 'y', 'z'])
+    check_type('origin', origin, Iterable, Real)
+
+    # Define function to create a plane on given axis
+    def plane(axis, name, value):
+        cls = globals()['{}Plane'.format(axis.upper())]
+        return cls(name='{} {}'.format(name, axis),
+                   boundary_type=boundary_type,
+                   **{axis + '0': value})
+    def cylinder(axis, name, ax1, val1, ax2, val2, r):
+        cls = globals()['{}Cylinder'.format(axis.upper())]
+        return cls(name='{} {}'.format(name, axis),
+                   boundary_type=boundary_type,
+                   **{ax1 + '0': val1,
+                      ax2 + '0': val2,
+                      'r': r})
+
+    if axis == 'x':
+        x1, x2 = 'y', 'z'
+        axcoord = 0
+        axcoord1 = 1
+        axcoord2 = 2
+    elif axis == 'y':
+        x1, x2 = 'x', 'z'
+        axcoord = 1
+        axcoord1 = 0
+        axcoord2 = 2
+    else:
+        x1, x2 = 'x', 'y'
+        axcoord = 2
+        axcoord1 = 0
+        axcoord2 = 1
+
+    # Get torus class corresponding to given axis
+    tor = globals()['{}Torus'.format(axis.upper())]
```

Comment:
You only need this if you are implementing the rounded edge/fillet, so perhaps make it conditional and/or move it closer to that code section.